quipucords/scanner/openshift/api.py

```diff
@@ -56,6 +58,30 @@ def _normalize_kwargs(kwargs):
     return _decorator
 
 
+def optional_openshift_resource(resource_name: str, empty_return_type: type = list):
+    """
+    Handle ResourceNotFoundError if the host does not support the given resource.
+
+    Returns an empty list if the resource is not found.
+    """
+
+    def _optional_openshift_resource(func):
+        @wraps(func)
+        def _wrapped_func(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except ResourceNotFoundError:
+                logger.info(
+                    _("This OpenShift host does not have %(resource_name)s."),
+                    {"resource_name": resource_name},
+                )
+                return empty_return_type()
```

Comment:
I couldn't follow where this function is 🤔

Reply:
`empty_return_type` is the type argument passed into the outer-most layer of the decorator. By default, it's `list`. So, this line by default means `return list()` or effectively `return []`.

Writing decorators that take arguments is always a minor nightmare for me as I struggle to remember (and have to search online to confirm) how many and _which_ nested layers need to do each thing. 😬 It does work, though! And it's consistent with a few decorators I've written in the past.